Update modules and durable objects docs to reflect recent changes (#1238)
cloudflare/cloudflare-docs: products/workers/src/content/cli-wrangler/configuration.md

- `usage_model` <Type>top level</Type> <PropMeta>optional</PropMeta>
  - Specifies the [Usage Model](/platform/pricing#usage-models) for your Worker. There are two options - [`bundled`](/platform/limits#bundled-usage-model) and [`unbound`](/platform/limits#unbound-usage-model). For newly created Workers, if the Usage Model is omitted it will be set to the [default Usage Model set on the account](https://dash.cloudflare.com/?account=workers/default-usage-model). For existing Workers, if the Usage Model is omitted, it will be set to the Usage Model configured in the dashboard for that Worker.

- `build` <Type>inherited</Type> <PropMeta>optional</PropMeta>

>   - Allows configuring a custom build step to be run by wrangler when building your worker. See the [custom builds documentation](configuration#build) for more details

Reviewer: ```suggestion
  - Allows configuring a custom build step to be run by wrangler when building your worker. See the [custom builds documentation](configuration#build) for more details.
```

</Aside>

#### Modules

Cloudflare Workers now supports uploading scripts as a collection of modules, instead of a single file. Scripts uploaded in this fashion are written a bit differently -- they `export` their event handlers instead of registering them using the global `addEventListener('fetch', handler...)`. Modules also receive bindings (like KV Namespaces, Config Vargs, and Secrets) as arguments to their handlers, rather than global variables. Uploaded modules can `import` (for ES Modules) or `require()` (for CommonJS modules) other uploaded modules.

<Aside>

>   **Note:** You'll need to opt-in to the [Durable Objects](/learning/using-durable-objects) open beta, to have access to the `modules` format, since it's required to use Durable Objects.

Reviewer 2: The comma after "beta" is unnecessary. And saying that modules are required to use Durable Objects isn't really a justification for requiring being in the Durable Objects beta to use modules, so I'd probably rephrase this more like

```
You currently need to opt-in to the [Durable Objects](/learning/using-durable-objects) open beta to be able to use the `modules` format. This restriction will be removed as modules are better supported in the Workers Dashboard.
```

</Aside>

```js
import html from './index.html'

export default {
  // request is almost the same as `event.request` from the service worker format, except for:
  // * waitUntil() and passThroughOnException() are accessible from ctx instead
  // * request.cf is currently not available
  // env is where bindings like KV namespaces, Durable Object namespaces, Config variables, and Secrets

>   // will be exposed under, instead of them being placed in global scope.

Reviewer: ```suggestion
  // are exposed, instead of them being placed in global scope.
```
